aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -402,6 +417,58 @@ public synchronized void initialize(
         });
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void query(
+            @NonNull String modelName,
```

Comment:
Can you just perform a lookup, get the class type, and skip all the reimplementation?